frontend/src/pages/article/components/article-form/libs/components/article-cover-upload/article-cover-upload.tsx

```diff
@@ -0,0 +1,143 @@
+  const [previewUrl, setPreviewUrl] = useState<string | null>(
+    initialPreviewUrl ?? null,
+  );
+  const [isDragActive, setIsDragActive] = useState(false);
+  const [isUploadFileLoading, setIsUploadFileLoading] = useState(false);
+
+  const error = errors[name]?.message;
+  const hasError = Boolean(error);
+
+  const handleUploadArticleCover = (
+    event_: ChangeEvent<HTMLInputElement>,
+  ): void => {
+    const [image] = event_.target.files ?? [];
+
+    if (image) {
+      const formData = new FormData();
+      formData.append('file', image);
+
+      setIsUploadFileLoading(true);
+      void dispatch(filesActions.uploadFile(formData))
+        .unwrap()
+        .then((fileData) => {
+          field.onChange(fileData.id);
+          setPreviewUrl(fileData.url);
+        })
+        .catch((error: Error) => {
+          appActions.notify({ type: 'error', message: error.message });
+          field.onChange(null);
+        })
+        .finally(() => {
+          setIsUploadFileLoading(false);
+        });
+    }
+  };
+
+  const handleRemoveCover = (): void => {
+    setPreviewUrl(null);
+    field.onChange(null);
+  };
+
+  const handleDragEnter = (): void => {
+    setIsDragActive(true);
+  };
+
+  const handleDragLeave = (): void => {
+    setIsDragActive(false);
+  };
+
+  const handleDrop = (): void => {
+    setIsDragActive(false);
+  };
+
+  return (
+    <>
+      <div className={styles.uploadWrapper}>
+        <div className={styles.header}>
+          <p className={styles.uploadTitle}>Add a cover image</p>
+        </div>
+        {isUploadFileLoading && <DotLoader isLoading />}
+        {previewUrl && (
+          <div className={styles.coverContainer}>
+            <div className={styles.removeCoverOverlay}>
+              <IconButton
+                iconName="trashBin"
+                className={styles.removeButton}
+                onClick={handleRemoveCover}
+              />
+            </div>
+            <img alt="article cover" src={previewUrl} />
+          </div>
+        )}
+        {!previewUrl && !isUploadFileLoading && (
+          <div
+            className={getValidClassNames(
+              styles.dropArea,
+              isDragActive && styles.dropAreaActive,
+            )}
+          >
+            <input
+              type="file"
+              onDrop={handleDrop}
+              className={styles.fileInput}
+              onDragEnter={handleDragEnter}
+              onDragLeave={handleDragLeave}
+              onChange={handleUploadArticleCover}
+              accept={SUPPORTED_FILE_TYPES_STRING}
+            />
+            <p className={styles.dropAreaHint}>
+              <Icon iconName="image" className={styles.imageIcon} /> Drag and
+              drop an image here or
+              <span className={styles.chooseImageButton}>choose image</span>
+            </p>
+          </div>
+        )}
+      </div>
+      {hasError && <ErrorMessage error={error as string} />}
```

Comment:
This way in case when error occurs - section with error message won't jump (because it's height would be changed from 0 (when there is no error), to actual error message height -- it will just have a static height 
```suggestion
      <ErrorMessage error={error as string} />
```